### PR TITLE
Iteration 3 non-working

### PR DESCRIPTION
**Pull request by normanrs**
I wasted a total of a full hour trying to diagnose two errors. Extremely frustrated.

### CODE REVIEW

File: lib/message.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class Message
+  attr_reader :text, :translator
+
+  def initialize(text, translator)
+    @text  = text
+    @translator   = translator
+  end
+
+  def translation
+    morse_translator.translate(@text)
```
Comment (s-espinosa) [inline lib/message.rb:12]: Close here. `translate` is a method on `MorseTranslator`, but you're holding your instance of `MorseTranslator` in the instance variable `@translator` here. In order to make this work you need to reference that instance variable.

File: lib/morse_translator.rb [hunk 2]
```diff
@@ -40,4 +41,12 @@ def initialize
                     " " => " "}
   end
 
+  def translate(english)
+    in_string = english.downcase.chars
+    out_string = []
+    in_string.map do |letter|
```
Comment (s-espinosa) [inline lib/morse_translator.rb:47]: If you're going to use map, don't need the `out_string` aggregator. That pattern is intended to be used with `.each`, when you either don't have access to `map` (for example in a language other than Ruby), or aren't aware of a more specific enumerable.

That said, like that you were able to get this to work!

File: test/message_test.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def test_translate
+    morse_translator = MorseTranslator.new
+    newmessage = Message.new("There are 3 ships", morse_translator)
+    # binding.pry
+    # message.translation
+    newmessage.translation
+    expected = "-......-.. .-.-.. ...-- ..........--...."
+    assert_equal expected, actual
```
Comment (s-espinosa) [inline test/message_test.rb:21]: Like this pattern in your tests.

File: test/morse_translator_test.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def test_translate
+    morse_translator = MorseTranslator.new
+    actual = morse_translator.translate("hello world")
+    expected = "......-...-..--- .-----.-..-..-.."
+    assert_equal expected, actual
+  end
+
+  def test_translate_capitals_and_numbers
```
Comment (s-espinosa) [inline test/morse_translator_test.rb:19]: Would be good to see a test for capitals separately as well.

**Review by s-espinosa — COMMENTED** [review]
@normanrs Generally on the right track! Comments below. Let us know if you have any questions. Thanks!

File: test/morse_translator_test.rb (hunk 4 above)
Comment (normanrs) [reply test/morse_translator_test.rb:19]: Totally agree

File: lib/morse_translator.rb (hunk 2 above)
Comment (normanrs) [reply lib/morse_translator.rb:47]: This one was a time issue

File: lib/message.rb (hunk 1 above)
Comment (normanrs) [reply lib/message.rb:12]: I've been hesitant to overuse instance variables in an effort to encapsulate my programs. I'm close, though, to understanding these scope issues.

**Comment by normanrs** [comment]
I wasted a total of almost an hour on two cases of errors that I couldn't figure out. One was a basic syntax issue the other was a misunderstanding of how methods are called in a test file. I learned from the latter. But overall I'm concerned about coding on my own because I really benefit from having a second set of eyes to just remind me of small things and talk things through.

**Pull request closed without merging**